helpers/sdxl/save_hooks.py

```diff
@@ -106,79 +113,85 @@ def __init__(
-            unet_lora_layers_to_save = None
-            transformer_lora_layers_to_save = None
-            text_encoder_1_lora_layers_to_save = None
-            text_encoder_2_lora_layers_to_save = None
-            text_encoder_3_lora_layers_to_save = None
-
-            for model in models:
-                if isinstance(model, type(unwrap_model(self.accelerator, self.unet))):
-                    unet_lora_layers_to_save = convert_state_dict_to_diffusers(
+        for model in models:
+            if isinstance(model, type(unwrap_model(self.accelerator, self.unet))):
+                unet_lora_layers_to_save = convert_state_dict_to_diffusers(
+                    get_peft_model_state_dict(model)
+                )
+            elif isinstance(
+                model, type(unwrap_model(self.accelerator, self.text_encoder_1))
+            ):
+                text_encoder_1_lora_layers_to_save = (
+                    convert_state_dict_to_diffusers(
                         get_peft_model_state_dict(model)
                     )
-                elif isinstance(
-                    model, type(unwrap_model(self.accelerator, self.text_encoder_1))
-                ):
-                    text_encoder_1_lora_layers_to_save = (
-                        convert_state_dict_to_diffusers(
-                            get_peft_model_state_dict(model)
-                        )
-                    )
-                elif isinstance(
-                    model, type(unwrap_model(self.accelerator, self.text_encoder_2))
-                ):
-                    text_encoder_2_lora_layers_to_save = (
-                        convert_state_dict_to_diffusers(
-                            get_peft_model_state_dict(model)
-                        )
+                )
+            elif isinstance(
+                model, type(unwrap_model(self.accelerator, self.text_encoder_2))
+            ):
+                text_encoder_2_lora_layers_to_save = (
+                    convert_state_dict_to_diffusers(
+                        get_peft_model_state_dict(model)
                     )
-                elif isinstance(
-                    model, type(unwrap_model(self.accelerator, self.text_encoder_3))
-                ):
-                    text_encoder_3_lora_layers_to_save = (
-                        convert_state_dict_to_diffusers(
-                            get_peft_model_state_dict(model)
-                        )
+                )
+            elif isinstance(
+                model, type(unwrap_model(self.accelerator, self.text_encoder_3))
+            ):
+                text_encoder_3_lora_layers_to_save = (
+                    convert_state_dict_to_diffusers(
+                        get_peft_model_state_dict(model)
                     )
+                )
 
-                elif isinstance(
+            elif not isinstance(model, type(unwrap_model(self.accelerator, HunyuanDiT2DModel))):
+                if isinstance(
```

Comment:
we go from `if isinstance()` to `elif not isinstance()` ? it looks like we never save the hunyuan-dit weights

Reply:
Because the codepath below is for LoRA and Hunyuan DiT doesn’t support LoRA yet. Or am I reading it wrong?

Reply:
since you'd refactored it, i assumed there was support :P